src/Utils/DOM.purs

```diff
@@ -52,21 +53,31 @@ foreign import getOffsetClientRect ∷ ∀ eff.  HTMLElement → Eff (dom ∷ DO
 foreign import open ∷ ∀ eff. String → String → String → Window → Eff (dom ∷ DOM | eff) Unit
 foreign import close ∷ ∀ eff. Window → Eff (dom ∷ DOM | eff) Unit
 foreign import centerPopupWindowFeatures ∷ ∀ eff. Int → Int → Window → Eff (dom ∷ DOM | eff) String
+foreign import setFontSize ∷ ∀ eff. HTMLElement → String → Eff (dom ∷ DOM | eff) Unit
+foreign import getOffsetWidth ∷ ∀ eff. HTMLElement → Eff (dom ∷ DOM | eff) Int
+foreign import getOffsetHeight ∷ ∀ eff. HTMLElement → Eff (dom ∷ DOM | eff) Int
+foreign import getScrollWidth ∷ ∀ eff. HTMLElement → Eff (dom ∷ DOM | eff) Int
+foreign import getScrollHeight ∷ ∀ eff. HTMLElement → Eff (dom ∷ DOM | eff) Int
 
 -- | Same as `getTextWidth` but w/o Eff wrapper. This function definitely has effects
 -- | of allocating canvas and should have `Eff (ref ∷ REF|e)` or `Eff (dom ∷ DOM|e)`
 -- | but since we don't use intermediate `canvas` anywhere it's safe to think about
 -- | this as pure function from font style and string to width.
 foreign import getTextWidthPure ∷ String → String → Number
 
--- | Takes an array of (div) elements, array of available font sizes, maximum dimension and
--- | returns sizes of fitted elements
-foreign import fitTexts
-  ∷ ∀ e
-  . Array HTMLElement
-  → Array Int
-  → {width ∷ Int, height ∷ Int}
-  → Eff (dom ∷ DOM|e) (Array {width ∷ Int, height ∷ Int})
+fits ∷ ∀ eff. HTMLElement → Eff (dom ∷ DOM | eff) Boolean
+fits el = (&&) <$> fitsHorizontally <*> fitsVertically
+  where
+  fitsHorizontally = (<=) <$> getScrollWidth el <*> getOffsetWidth el
+  fitsVertically = (<=) <$> getScrollHeight el <*> getOffsetHeight el
+
+fitText ∷ ∀ eff. Array Int → HTMLElement → Eff (dom ∷ DOM | eff) (Maybe Int)
```

Comment:
I don't think so. This probably should be implemented via `tailRec`. But `fontSizes` shouldn't be very large.